src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,150 @@
+                }
+            }
+
+            var entries = new Entries(bundles);
+            return Option.Some(entries);
+        }
+
+        private static bool WithinRange(DateRange range, DateTime date)
+        {
+            var fromDate = ParseDate(range.From);
+            var toDate = ParseDate(range.To);
+            return date >= fromDate && date < toDate;
+        }
+
+        private static DateTime ParseDate(string dateString)
+        {
+            var formatStrings = new[]
+            {
+                "yyyy-MM-dd", "yyyy-MM-dd hh:mm:ss", "yyyy-MM-dd hh:mm:ss tt", "yyyy-MM-ddTHH:mm:ss.fffzzz",
+                "dd/MM/yyyy", "dd/MM/yyyy hh:mm:ss", "dd/MM/yyyy hh:mm:ss tt", "dd/MM/yyyyTHH:mm:ss.fffzzz"
+            };
+            var tryParseExact = DateTime.TryParseExact(dateString,
+                formatStrings,
+                CultureInfo.CurrentCulture,
+                DateTimeStyles.None,
+                out var aDateTime);
+            if (!tryParseExact)
+            {
+                Log.Error($"Error parsing date: {dateString}");
+            }
+
+            return aDateTime;
+        }
+
+        private async Task<Dictionary<string, List<HealthObservationRepresentation>>> FindPatientData(
+            DataRequest dataRequest)
+        {
+            LogDataRequest(dataRequest);
+            var caseId = dataRequest.CareContexts.First().PatientReference;
+            var patient = _patientRepository.PatientWith(caseId);
+            var patientName = patient.Select(patient => patient.Name).ValueOr("");
+
+            var careContextsAndListOfDataFiles = new Dictionary<string, List<HealthObservationRepresentation>>();
+            var request = new HttpRequestMessage(HttpMethod.Post, "https://tmc.gov.in/tmh_ncg_api/healthInfo")
+            {
+                Content = new StringContent(
+                    JsonConvert.SerializeObject(new
+                    {
+                        HiTypes = dataRequest.HiType,
+                        StartDate = dataRequest.DateRange.From,
+                        EndDate = dataRequest.DateRange.To,
+                        CaseId = caseId
+                    }),
+                    Encoding.UTF8,
+                    "application/json")
+            };
+
+            var response = await _client.SendAsync(request);
+            await using var responseStream = await response.Content.ReadAsStreamAsync();
+            var clinicalNotes = await JsonSerializer.DeserializeAsync<IEnumerable<ClinicalNote>>(responseStream);
+            var uuid = Uuid.Generate().Value;
+            var id = uuid.Split(":").Last();
+            var healthObservationRepresentations = new List<HealthObservationRepresentation>();
+            foreach (var clinicalNote in clinicalNotes)
+            {
+                if (!WithinRange(dataRequest.DateRange, clinicalNote.CreatedDate)) continue;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)